app/controllers/course/forum/topics_controller.rb

```diff
@@ -10,10 +10,8 @@ class Course::Forum::TopicsController < Course::Forum::ComponentController
   before_action :add_topic_breadcrumb
 
   def show
-    @posts = @topic.posts.ordered_by_created_at
```

Comment:
@lowjoel I think this scope still need to be adopted, this makes sure that the original order is correct before applying the topological sort.